ext/opentelemetry-ext-datadog/src/opentelemetry/ext/datadog/__init__.py

```diff
@@ -0,0 +1,211 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The **OpenTelemetry Datadog Exporter** provides a span exporter from
+`OpenTelemetry`_ traces to `Datadog`_ by using the Datadog Agent.
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry import trace
+    from opentelemetry.ext.datadog import DatadogSpanExporter
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+
+    trace.set_tracer_provider(TracerProvider())
+    tracer = trace.get_tracer(__name__)
+
+    exporter = DatadogSpanExporter(
+        agent_url="http://agent:8126", service="my-helloworld-service"
+    )
+
+    span_processor = BatchExportSpanProcessor(exporter)
+    trace.get_tracer_provider().add_span_processor(span_processor)
+
+    with tracer.start_as_current_span("foo"):
+        print("Hello world!")
+
+API
+---
+.. _Datadog: https://www.datadoghq.com/
+.. _OpenTelemetry: https://github.com/open-telemetry/opentelemetry-python/
+"""
+
+import logging
+import os
+from urllib.parse import urlparse
+
+from ddtrace.encoding import MsgpackEncoder
+from ddtrace.internal.writer import AgentWriter
+from ddtrace.span import Span as DatadogSpan
+
+import opentelemetry.trace as trace_api
+from opentelemetry.sdk.trace import Span
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.trace.status import StatusCanonicalCode
+
+log = logging.getLogger(__name__)
+
+DEFAULT_AGENT_URL = "http://localhost:8126"
+DATADOG_SPAN_TYPES = (
+    "sql",
+    "mongodb",
+    "http",
+)
+
+
+class DatadogSpanExporter(SpanExporter):
+    """Datadog span exporter for OpenTelemetry.
+
+    Args:
+        agent_url: The url of the Datadog Agent or use `DD_TRACE_AGENT_URL` environment variable
+        service: The service to be used for the application or use `DD_SERVICE` environment variable
+    """
+
+    def __init__(self, agent_url=None, service=None):
+        self.agent_url = (
+            agent_url
+            if agent_url
+            else os.environ.get("DD_TRACE_AGENT_URL", DEFAULT_AGENT_URL)
+        )
+        self.service = service if service else os.environ.get("DD_SERVICE")
+        self._agent_writer = None
+
+    @property
+    def agent_writer(self):
+        if self._agent_writer is None:
+            url_parsed = urlparse(self.agent_url)
```

Comment:
:+1: So much cleaner than in `ddtrace` where we support providing the hostname, port _and_ url. 😆

Reply:
👏